### PR TITLE
bpo-36440: include node names in ParserError messages, instead of numeric IDs

### PR DESCRIPTION
**Pull request by tyomitch**
https://bugs.python.org/issue36440

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2019-03-25-13-45-19.bpo-36440.gkvzhi.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Include node names in ParserError messages, instead of numeric IDs
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2019-03-25-13-45-19.bpo-36440.gkvzhi.rst]: Can you add formatting to the NEWS entry (so ParserError will have a link in the rendered documentation). Also, add "Patch by A. Skrobov" at the end :)

File: Lib/test/test_parser.py [hunk 2]
```diff
@@ -749,6 +749,14 @@ def test_illegal_encoding(self):
         with self.assertRaises(UnicodeEncodeError):
             parser.sequence2st(tree)
 
+    def test_message(self):
```
Comment (pablogsal) [inline Lib/test/test_parser.py]: Maybe a more descriptive name for the test?

**Comment by pablogsal** [comment]
You have some trailing whitespace in `parsermodule.c`

**Review by serhiy-storchaka — COMMENTED** [review]
What if the index out of the boundary of the array?

**Comment by tyomitch** [comment]
> What if the index out of the boundary of the array?

Fixed

File: Modules/parsermodule.c [hunk 3]
```diff
@@ -666,6 +662,12 @@ validate_node(node *tree)
     for (pos = 0; pos < nch; ++pos) {
         node *ch = CHILD(tree, pos);
         int ch_type = TYPE(ch);
+        if ((ch_type >= NT_OFFSET + _PyParser_Grammar.g_ndfas)
```
Comment (serhiy-storchaka) [inline Modules/parsermodule.c:665]: What if `ch_type` is negative?

This guards `ch_type`, but what about `next_type`?

Comment (tyomitch) [reply Modules/parsermodule.c:665]: > What if `ch_type` is negative?

Good point! Now fixed.

> This guards `ch_type`, but what about `next_type`?

`next_type` is taken from `_PyParser_Grammar`, which is hardcoded in graminit.c; so, it's guaranteed to be valid.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by tyomitch** [comment]
@pablogsal do you have any unresolved comments regarding this patch?

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
@tyomitch Thanks for the ping and sorry for the delay!

**Pull request merged**